python/Ganga/GPIDev/Lib/Registry/RegistrySlice.py

```diff
@@ -368,21 +373,17 @@ def __getitem__(self, x):
                 raise RegistryKeyError('object "%s" not found' % x)
             return addProxy(self.objects[ids[0]])
 
-        raise RegistryAccessError('Expected int or string (job name).')
+        if isinstance(x, slice):
+            start = x.start if x.start is not None else ''
+            stop = x.stop if x.stop is not None else ''
 
-    def __getslice__(self, i1, i2):
-        import sys
+            returnable = self.__class__("%s[%s:%s]" % (self.name, start, stop))
+            for id_ in self.ids()[x]:
+                returnable.objects[id_] = self.objects[id_]
 
-        if i2 == sys.maxsize:
-            endrange = ''
-        else:
-            endrange = str(i2)
+            return addProxy(returnable)
```

Comment:
I'm avoiding changing things as much as possible in this. We can fix this in another PR as I'm wanting to focus on as fewer things as possible in this one.
These used to be a proxy applied here and this class does not make use of a lot of the standard proxy machinery so I'm not sure and don't want to rock the boat by changing the workflow in this PR.

(Arguably I should revert the changes I made to the imports but as that's PEP8 and relatively obvious the consequence I'll leave those changes in.)